### PR TITLE
260 remove classif

### LINKED ISSUE #260: Reimplement classification task
In current version of GDL, the only task available is [segmentation](https://github.com/NRCan/geo-deep-learning/blob/develop/config/task/segmentation.yaml)

PR #208 removed [train_classification](https://github.com/NRCan/geo-deep-learning/pull/208/files#diff-e949dabb684fb3241e7636304e7931e800a9bed11e9a767e27d6d38da2e19b91) and classification at inference [remains implemented in inference_segmentation.py](https://github.com/NRCan/geo-deep-learning/blob/develop/inference_segmentation.py#L612), but is unreachable due to the current usage of GDL.py using [mode and task to execute a specific script](https://github.com/NRCan/geo-deep-learning/blob/develop/config/task/segmentation.yaml#L2).

### PR DESCRIPTION
**Pull request by remtav**
Also:
replace load_obj() with hydra's get_method
remove task config directory

closes #260

### CODE REVIEW

File: config/task/segmentation.yaml [hunk 1]
```diff
@@ -1,2 +0,0 @@
-task_name: segmentation
-path_task_function: ${mode}_segmentation.main
```
Comment (remtav) [inline config/task/segmentation.yaml:2]: @CharlesAuthier let me know if there was a particular reason you wanted it like this.

File: data/images_to_samples_ci_csv.csv [hunk 2]
```diff
@@ -1,2 +1,2 @@
-./data/22978945_15.tif,,./data/massachusetts_buildings.gpkg,,trn
-./data/23429155_15.tif,,./data/massachusetts_buildings.gpkg,,tst
+./22978945_15.tif,,./massachusetts_buildings.gpkg,,trn
+./23429155_15.tif,,./massachusetts_buildings.gpkg,,tst
```
Comment (remtav) [inline data/images_to_samples_ci_csv.csv:2]: This is temporary. Will refactor the in_case_of_path parameter. I'd rather use hydra.utils.get_original_cwd().

File: utils/utils.py [hunk 3]
```diff
@@ -427,7 +424,7 @@ def try2read_csv(path_file, in_case_of_path, msg):
         Path(path_file).resolve(strict=True)
     except FileNotFoundError:
         if in_case_of_path:
-            path_file = os.path.join(in_case_of_path, os.path.basename(path_file))
+            path_file = str(Path(in_case_of_path) / (path_file.split('./')[-1]))
             try:
```
Comment (remtav) [inline utils/utils.py:428]: I'll get back to this in_case_of_path story in a future PR. Would like to suggest a different, lighter approach with hydra.utils.get_original_cwd()

File: utils/utils.py [hunk 4]
```diff
@@ -630,26 +627,6 @@ def compare_config_yamls(yaml1: dict, yaml2: dict, update_yaml1: bool = False) -
...
-    """
-    obj_path_list = obj_path.rsplit('.', 1)
-    obj_path = obj_path_list.pop(0) if len(obj_path_list) > 1 else default_obj_path
-    obj_name = obj_path_list[0]
-    module_obj = imp.import_module(obj_path)
-    if not hasattr(module_obj, obj_name):
-        raise AttributeError(f"Object `{obj_name}` cannot be loaded from from `{obj_path}`.")
-    return getattr(module_obj, obj_name)
-
-
```
Comment (remtav) [inline utils/utils.py:652]: @CharlesAuthier same here. Let me know if you had anytning in my mind for further developments with this function. If the purpose was just to use in GDL.py to reach the right method, I think hydra.utils.get_method() does the job.

**Review by mpelchat04 — APPROVED** [review]

**Comment by CharlesAuthier** [comment]
you have a conflict, after change for it I will approve

File: inference_segmentation.py [hunk 5]
```diff
@@ -623,70 +549,56 @@ def main(params: dict) -> None:
...
+    # TODO: Add verifications?
+    if bucket:
+        bucket.download_file(loaded_checkpoint, "saved_model.pth.tar")  # TODO: is this still valid?
+        model, _ = load_from_checkpoint("saved_model.pth.tar", model)
+    else:
+        model, _ = load_from_checkpoint(loaded_checkpoint, model)
 
-        # TODO: Add verifications?
+    # Save tracking TODO put option not just mlflow
+    if 'tracker_uri' in locals() and 'run_name' in locals():
```
Comment (remtav) [inline inference_segmentation.py:563]: No major change here, git's just having a hard time following the movement of big blocks of code.

File: inference_segmentation.py [hunk 6]
```diff
@@ -623,118 +549,113 @@ def main(params: dict) -> None:
...
-            all_gdf = pd.concat(gdf_)  # Concatenate all geo data frame into one geo data frame
-            all_gdf.reset_index(drop=True, inplace=True)
-            gdf_x = gpd.GeoDataFrame(all_gdf)
-            bench_gpkg = working_folder / "benchmark.gpkg"
-            gdf_x.to_file(bench_gpkg, driver="GPKG", index=False)
-            logging.info(f'\nSuccessfully wrote benchmark geopackage to: {bench_gpkg}')
-        # log_artifact(working_folder)
\ No newline at end of file
+                                     out_shape=(inf_meta['height'], inf_meta['width']),
+                                     attribute_name=attribute_field,
```
Comment (remtav) [inline inference_segmentation.py:604]: Same here. Nothing major has happened. Just deindented after removing if/else for classif/segmentation task.

**Review by victorlazio109 — APPROVED** [review]

**Review by mpelchat04 — APPROVED** [review]

**Review by CharlesAuthier — APPROVED** [review]

**Pull request merged**